packages/react-vapor/src/components/form/FormGroup.tsx

```diff
@@ -1,16 +1,34 @@
 import * as classNames from 'classnames';
 import * as React from 'react';
 
+type FormGroupMods = 'mod-header-padding' | 'mod-form-top-bottom-padding' | 'material-card';
+
 export interface IFormGroupProps {
-    title: string;
-    description?: string;
+    title?: string;
+    description?: React.ReactNode;
     className?: string;
+    mods?: FormGroupMods | FormGroupMods[];
+    level?: 1 | 2 | 3;
 }
 
-export const FormGroup: React.FunctionComponent<IFormGroupProps> = ({children, title, description, className}) => (
-    <div className={classNames(className, 'form-group', 'mod-padding-children')}>
-        <h3 className="text-medium-blue bold">{title}</h3>
-        {description && <p className="description">{description}</p>}
-        {children}
-    </div>
-);
+export const FormGroup: React.FunctionComponent<IFormGroupProps> = ({
+    children,
+    title,
+    description,
+    className,
+    mods,
+    level,
+}) => {
+    const titleProps = {
+        className: 'text-medium-blue mb2',
+        children: title,
+    };
+    const levelElement = [() => <h2 {...titleProps} />, () => <h3 {...titleProps} />, () => <h4 {...titleProps} />];

```

Comment:
I think something like this would be easier to read.
```tsx
const H = level ? `h${level}` : 'h1';
return (
    ...
    {title && <H {...titleProps} />}
);
```

Reply:
> Is there something we can do about the accumulated spacing top?

You're right, I had a wrong CSS rule check the demo back with the latest commit

Reply:
And TIL about dynamic name tag, amazing stuff there